kubernetes/kind.go

```diff
@@ -0,0 +1,81 @@
+package kubernetes
+
+import (
+	_ "embed"
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/aws/ec2/ec2"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	kindVersion       = "v0.17.0"
+	kindReadinessWait = "60s"
+)
+
+//go:embed kind-cluster.yaml
+var kindClusterConfig string
+
+// Install Kind on a Linux virtual machine
+// Currently using ec2.VM waiting for correct abstraction
+func NewKindCluster(vm *ec2.VM, clusterName, arch string) (*remote.Command, error) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
I wonder if we should rather have installer helpers in the VM, so having for example a `KubernetesManager` like we do have a `FileManager`. 

Installing Kind would look like 

```golang
vm.KubernetesManager.CreateKindCluster(clusterName, arch string)
```

This way we could prevent having a public `Runner` in the VM

Reply:
Same for `DockerManager`, it could `vm.DockerManager.InstallAgent(version, image, extraCompose string)`

What do you think ?

Reply:
What I probably find complex in `Command` is the requirement for a `delete` command. There are many parameters that might be counter intuitives. 

Do we expect test code to use the `Runner` ? 

I see your point, agree having all managers would create lot of coupling on the VM.

Reply:
It's not a requirement at all, it can be `nil` if it makes sense.